### PR TITLE
Misc fixes and improvements

### PR DESCRIPTION
**Pull request by ramereth**
This includes a variety of fixes and improvements that I feel would help this cookbook. This should probably result in a
major cookbook bump.

- Improve customization of server with additional attributes
- Set `sensitive true` for `slapd.conf` template
- Fix EL8 systemd unit for slapd daemon
- Move platform attributes to library helpers
- Move resource methods to library helpers
- Add `install_client` and `install_server` properties to `openldap_install` resource
- Fedora also needs its systemd unit file adjusted
- Improve ChefSpec tests
- Automatically rebuild slapd.d configuration when slapd.conf is updated
- Remove unused EL6 template
- Documentation for openldap_install resource

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: recipes/default.rb [hunk 1]
```diff
@@ -21,42 +21,40 @@
...
 ##  Set syncrepl_consumer_config dynamic values here
 node.default_unless['openldap']['syncrepl_consumer_config']['searchbase'] = "\"#{node['openldap']['basedn']}\""
 node.default_unless['openldap']['syncrepl_consumer_config']['binddn'] = "\"#{node['openldap']['syncrepl_cn']},#{node['openldap']['basedn']}\""
 node.default_unless['openldap']['syncrepl_consumer_config']['credentials'] = "\"#{node['openldap']['slapd_replpw']}\""
 
-template "#{node['openldap']['dir']}/slapd.conf" do
+systemd_unit 'slapd.service' do
+  content openldap_el8_systemd_unit
+  action [:create]
+end if openldap_el8_systemd_unit?
```
Comment (detjensrobert) [inline recipes/default.rb:37]: Should this (& others) be `only_if` guards instead?

**Review by detjensrobert — CHANGES_REQUESTED** [review]

File: recipes/default.rb (hunk 1 above)
Comment (ramereth) [reply recipes/default.rb:37]: Doing it this way will allow the resource not to show up in the chef runs (even as not doing anything). I believe this has been an standard way of doing things in Sous Chefs as of late but I might be wrong.

**Review by detjensrobert — APPROVED** [review]
LGTM

**Comment by kitchen-porter** [comment]
Released as: [5.0.0](https://github.com/sous-chefs/openldap/releases/tag/5.0.0)

**Pull request merged**